src/litserve/server.py

```diff
@@ -295,6 +317,55 @@ async def data_reader():
 
             return data
 
+        @self.app.post("/stream-predict", dependencies=[Depends(setup_auth())])
+        async def stream_predict(request: self.request_type, background_tasks: BackgroundTasks) -> self.response_type:
+            uid = uuid.uuid4()
+
+            read, write = self.new_pipe()
+
+            if self.request_type == Request:
+                self.app.request_buffer[uid] = (await request.json(), write)
+            else:
+                self.app.request_buffer[uid] = (request, write)
+
+            self.app.request_queue.put(uid)
+            background_tasks.add_task(cleanup, self.app.request_buffer, uid)
+
+            async def event_wait(evt, timeout):
+                # suppress TimeoutError because we'll return False in case of timeout
+                with contextlib.suppress(asyncio.TimeoutError):
+                    await asyncio.wait_for(evt.wait(), timeout)
+                return evt.is_set()
+
+            async def stream_from_pipe():
+                while True:
+                    if read.poll(self.app.timeout):
+                        data = read.recv()
+                        if isinstance(data, StopIteration):
+                            return
+                        yield data
+                    asyncio.sleep(0.01)
```

Comment:
Had to do this for Windows :(. Any suggestion @lantiga?

Reply:
It is what it is. But let's make sure it's clear that this codepath will only be followed on windows (add a comment or assert outside the while loop)

Reply:
In any case 10ms sleep is a lot while streaming, we need to decrease it by an order of magnitude.